plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/refactoring/RefactorElementNameFragment2.xtend

```diff
@@ -0,0 +1,149 @@
+/*******************************************************************************
+ * Copyright (c) 2015 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.xtext.generator.ui.refactoring
+
+import com.google.inject.Inject
+import com.google.inject.name.Names
+import org.eclipse.xtend.lib.annotations.Accessors
+import org.eclipse.xtext.Grammar
+import org.eclipse.xtext.xtext.generator.AbstractGeneratorFragment2
+import org.eclipse.xtext.xtext.generator.XtextGeneratorNaming
+import org.eclipse.xtext.xtext.generator.model.GuiceModuleAccess
+import org.eclipse.xtext.xtext.generator.xbase.XbaseUsageDetector
+
+import static extension org.eclipse.xtext.GrammarUtil.*
+import static extension org.eclipse.xtext.xtext.generator.model.TypeReference.*
+
+/**
+ * Contributes the registration of element renaming infrastructure.
+ *  
+ * @author Christian Schneider - Initial contribution and API
+ */
+class RefactorElementNameFragment2 extends AbstractGeneratorFragment2 {
+
+	@Inject
+	extension XtextGeneratorNaming
+
+	@Inject
+	extension XbaseUsageDetector
+
+	@Accessors
+	private Boolean useJdtRefactoring = null; // no default (depends on whether xbase is used or not)	
```

Comment:
The comment could be JavaDoc and explicitly state that it defaults to 'false' if Xbase is not used and otherwise to true.